torch/_functorch/partitioners.py

```diff
@@ -314,10 +314,13 @@ def _size_of(node: fx.Node) -> int:
                 return 1
             else:
                 return 999999
+        # NB: The fallback values here are meaningless, maybe we should respect
+        # torch._inductor.config.unbacked_symint_fallback (but this is a
+        # layering violation)
```

Comment:
Instead of picking a random default, should we the value ranges upper bound if one exists? (I think it's something like `hint_int(shape_env.var_to_range[val.node.expr].upper)`)?

Also, just curious since you're picking a random value here - do you think this value won't matter in practice? (that we are unlikely to run into models where some tensor with an unbacked size could optionally be saved in the bw vs recomputed). Do you think we should just introduce a new config to avoid the layering issue?
```
torch._functorch.config.unbacked_symint_fallback = 8192
torch._inductor.config.unbacked_symint_fallback = torch._functorch.config.unbacked_symint_fallback
```

Reply:
> Instead of picking a random default, should we the value ranges upper bound if one exists? (I think it's something like hint_int(shape_env.var_to_range[val.node.expr].upper))?

We should. Doing this in general is somewhat involved, so we should probably do it in its own PR.

> Also, just curious since you're picking a random value here - do you think this value won't matter in practice? (that we are unlikely to run into models where some tensor with an unbacked size could optionally be saved in the bw vs recomputed).

I think it can eventually matter in complicated enough situations. But there are some countervailing factors: (1) we can also make suboptimal decisions when hints do exist--if we compile with a hint that is not representative of the true cost of something, and (2) if the user doesn't warmup on representative inputs, the correct range here is fundamentally unknowable. Even the upper bound, if it exists, is merely that: an upper bound; it may be that you expect it to be half the size of the theoretical upper bound, and that will make a big difference. So in the end, to handle this correctly, we will need some sort of user annotation or profile-guided optimization approach. So I'm not *too* worried about this as a stopgap right now.

> Do you think we should just introduce a new config to avoid the layering issue?

Maybe. Actually I think @peterbell10 added a new dedicated ShapeEnv config at https://github.com/pytorch/pytorch/pull/112933 so we can probably just centralize it all there.